### PR TITLE
implemented props from figma , and add some helpers for responsive design  , classes and breakpoints

### PR DESCRIPTION
**Pull request by TheRankN1**
https://trello.com/c/XwkjXxoE/58-component-shop-header

### CODE REVIEW

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<div>
+
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.html:2]: empty line

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="header-main">
+        <img class="logo center-on-mobile" src="../../../../../assets/logo.png" alt="" />
+        <div class="search">
+            <div class="all-categories hide-on-mobile">
+                <span>All categories</span>
+                <i class="demo-icon icon-arrow-down hide-on-mobile"></i>
+            </div>
+            <input type="text" placeholder="Search Products, categories ...">
+
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.html:11]: empty line

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            <i class="demo-icon icon-search"></i>
+        </div>
+        <div class="icons hide-on-mobile">
+            <i class="demo-icon icon-user"></i>
+            <i class="demo-icon icon-basket"></i>
+            <!-- <cavelizer-icon name="arrowDown"></cavelizer-icon> -->
+
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.html:18]: empty line

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+                <i class="demo-icon icon-arrow-down hide-on-mobile"></i>
+            </div>
+            <input type="text" placeholder="Search Products, categories ...">
+
+            <i class="demo-icon icon-search"></i>
+        </div>
+        <div class="icons hide-on-mobile">
+            <i class="demo-icon icon-user"></i>
+            <i class="demo-icon icon-basket"></i>
+            <!-- <cavelizer-icon name="arrowDown"></cavelizer-icon> -->
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.html:17]: de ce e asta commentat? decomenteaza-l / sterge-l

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+        </div>
+    </div>
+
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.html:21]: empty line

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+.header-main {
+  display: flex;
+  justify-content: space-between;
+
+  padding: 56px 45px;
+  .logo {
+    cursor: pointer;
+    padding: 16px;
+  }
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:11]: baga un enter dupa }

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    padding: 10px 16px;
+
+    border: 1px solid #d1d1d1;
+    border-radius: 12px;
+    width: 500px;
+
+    .all-categories {
+      font-weight: 700;
+      font-size: 15px;
+      line-height: 22px;
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:24]: baga un enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+      span {
+        margin-right: 8px;
+      }
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:27]: baga un enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+      i {
+        align-items: center;
+        color: $main-color;
+        font-size: 8px;
+        padding-right: 24px;
+
+        border-right: 1px solid #d1d1d1;
+      }
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:35]: baga un enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        cursor: pointer !important;
+        font-size: 11px;
+        color: $color--black;
+      }
+    }
+
+    input {
+      background-color: transparent;
+      border: none;
+      flex: 1;
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:46]: baga un enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+        font-size: 14px;
+        line-height: 19px;
+        color: $color--grey;
+        margin-left: 12px;
+        flex: 1;
+        padding-left: 24px;
+        display: block;
+        width: 400px;
+      }
+      &:focus {
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:60]: baga un enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        outline: none !important ;
+        text-align: center;
+        &::placeholder {
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:63]: baga un enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+          text-align: left;
+        }
+      }
+    }
+  }
+  .icons {
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:69]: enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    .icon-user {
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:73]: enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+      margin-right: 37px;
+      font-size: 18px;
+      cursor: pointer;
+    }
+    .icon-basket {
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:78]: enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+      position: relative;
+      cursor: pointer;
+      font-size: 18px;
+    }
+  }
+}
+@media (max-width: $mobile-breakpoint) {
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:85]: enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+  .header-main {
+    flex-direction: column;
+    .logo {
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:88]: enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+      width: 177px;
+      height: 24px;
+      margin: 0 auto;
+      box-sizing: content-box !important;
+    }
+    .search {
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:94]: enter

File: frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+      width: 75%;
+      margin: 0 auto;
+      .placeholder {
```
Comment (vestemeansorin91) [inline frontend/src/app/modules/shop/components/shop-header/shop-header.component.scss:97]: enter

File: frontend/src/app/shared/scss/_colors.scss [hunk 20]
```diff
@@ -1,6 +1,8 @@
 /* Variables */
 $main-color: #6a983c;
 $main-dark-color: #46760a;
+$color--black: #151515;
+$color--grey: #b6b6b6;
```
Comment (vestemeansorin91) [inline frontend/src/app/shared/scss/_colors.scss:5]: la variabile punem un singur dash, nu doua, doar la clase (cand avem modificatori) 
http://getbem.com/

File: frontend/src/app/shared/scss/_helpers.scss [hunk 21]
```diff
@@ -129,3 +129,24 @@
     display: none;
   }
 }
+/* RESPONSIVE HELPERS */
+.hide-on-mobile {
+  @media (max-width: $mobile-breakpoint) {
+    display: none !important;
+  }
+}
+.hide-on-tablet {
```
Comment (vestemeansorin91) [inline frontend/src/app/shared/scss/_helpers.scss:138]: enter

File: frontend/src/app/shared/scss/_helpers.scss [hunk 22]
```diff
@@ -129,3 +129,24 @@
...
+  @media (max-width: $tablet-breakpoint) {
+    display: none !important;
+  }
+}
+.center-on-mobile {
```
Comment (vestemeansorin91) [inline frontend/src/app/shared/scss/_helpers.scss:143]: enter

File: frontend/src/app/shared/scss/_helpers.scss [hunk 23]
```diff
@@ -129,3 +129,24 @@
...
+  @media (max-width: $mobile-breakpoint) {
+    text-align: center !important;
+  }
+}
+.hide-on-desktop {
```
Comment (vestemeansorin91) [inline frontend/src/app/shared/scss/_helpers.scss:148]: enter

File: frontend/src/app/shared/scss/_colors.scss [hunk 24]
```diff
@@ -1,6 +1,8 @@
 /* Variables */
 $main-color: #6a983c;
 $main-dark-color: #46760a;
```
Comment (vestemeansorin91) [inline frontend/src/app/shared/scss/_colors.scss:3]: schimba-le si pe astea in color-main si color-main-dark , ca sa pastram structura de 

color-ceva

**Review by mihaifirst — APPROVED** [review]

**Pull request merged**